test/data/javascript.js

```diff
@@ -104,6 +107,218 @@ exports.test_data = {
 
         ],
     }, {
+        name: "operator_position option - ensure no neswlines if preserve_newlines is false",
+        matrix: [
+            {
+                options: [
+                    { name: "operator_position", value: "'before_newline'" },
+                    { name: "preserve_newlines", value: "false" }
+                ]
+            }, {
+                options: [
+                    { name: "operator_position", value: "'after_newline'" },
+                    { name: "preserve_newlines", value: "false" }
+                ]
+            }, {
+                options: [
+                    { name: "operator_position", value: "'preserve_newline'" },
+                    { name: "preserve_newlines", value: "false" }
+                ]
+            }
+        ],
+        tests: [
+            {
+                unchanged: inputlib.operator_position.sanity
+            }, {
+                input: inputlib.operator_position.comprehensive,
+                output: inputlib.operator_position.sanity,
+            }
+        ]
+    }, {
+        name: "reset preserve_newlines and operator_position",
+        options: [
+            { name: "preserve_newlines", value: "true" },
+            { name: "operator_position", value: "'before_newline'" }
+        ]
+    }, {
+        name: "operator_position option - set to 'before_newline' (default value)",
+        tests: [
+            {
+                comment: 'comprehensive, various newlines',
+                input: inputlib.operator_position.comprehensive,
+                output: [
+                    'var res = a + b - c /',
+                    '    d * e %',
+                    '    f;',
```

Comment:
Peek: yeah, in the javascript module all tokens  are parsed before the formatting handlers run. Here's an example:
https://github.com/beautify-web/js-beautify/blob/master/js/lib/beautify.js#L749 

This is not the case in css or html modules - I haven't had time to untangle those.

As to behavior: I think I'm just sticking with what I outlined in https://github.com/beautify-web/js-beautify/pull/735#issuecomment-155311837 .   If I'm off target, please call me out.  It's been known to happen. :smile: